File: single-language-analysis/dataset-files/python/test_smell_101.py
# Original URL: https://github.com/Pham-Ngoc-Huy/Two_Sum/blob/268c021fc652ae99c817dd79dd1085d1de1593ee/automate/Lib/site-packages/pandas/tests/arrays/test_datetimes.py#L365-L389

class TestDatetimeArray:

    @pytest.mark.parametrize("dtype", ["datetime64[ns]", "datetime64[ns, UTC]"])
    @pytest.mark.parametrize(
        "other", ["datetime64[ns]", "datetime64[ns, UTC]", "datetime64[ns, CET]"]
    )
    def test_smell_101(self, dtype, other):
        # https://github.com/pandas-dev/pandas/pull/32490
        ser = pd.Series([1, 2], dtype=dtype)
        orig = ser.copy()

        err = False
        if (dtype == "datetime64[ns]") ^ (other == "datetime64[ns]"):
            # deprecated in favor of tz_localize
            err = True

        if err:
            if dtype == "datetime64[ns]":
                msg = "Use obj.tz_localize instead or series.dt.tz_localize instead"
            else:
                msg = "from timezone-aware dtype to timezone-naive dtype"
            with pytest.raises(TypeError, match=msg):
                ser.astype(other)
        else:
            t = ser.astype(other)
            t[:] = pd.NaT
            tm.assert_series_equal(ser, orig)

File: single-language-analysis/dataset-files/java/TestSmell101.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.Arrays;

public class TestSmell101 {

    @Test
    public void testAstypeCopies() {
        String[] dtypes = {"datetime64[ns]", "datetime64[ns, UTC]"};
        String[] others = {"datetime64[ns]", "datetime64[ns, UTC]", "datetime64[ns, CET]"};

        for (String dtype : dtypes) {
            for (String other : others) {
                Series ser = new Series(Arrays.asList(1, 2), dtype);
                Series orig = ser.copy();

                boolean err = false;
                if ((dtype.equals("datetime64[ns]") && !other.equals("datetime64[ns]")) || 
                    (!dtype.equals("datetime64[ns]") && other.equals("datetime64[ns]"))) {
                    err = true;
                }

                if (err) {
                    String msg;
                    if (dtype.equals("datetime64[ns]")) {
                        msg = "Use obj.tz_localize instead or series.dt.tz_localize instead";
                    } else {
                        msg = "from timezone-aware dtype to timezone-naive dtype";
                    }
                    try {
                        ser.astype(other);
                        fail("Expected TypeError not thrown");
                    } catch (TypeError e) {
                        assertEquals(msg, e.getMessage());
                    }
                } else {
                    Series t = ser.astype(other);
                    t.fillNaT();
                    assertSeriesEqual(ser, orig);
                }
            }
        }
    }

    private void assertSeriesEqual(Series expected, Series actual) {
        // Implement the logic to assert that two Series are equal
    }
}